WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListViewModel.swift

```diff
@@ -0,0 +1,107 @@
+        let config = BlogListConfiguration.defaultConfig
+        let dataSource = BlogListDataSource(contextManager: contextManager)
+        dataSource.shouldHideSelfHostedSites = config.shouldHideSelfHostedSites
+        dataSource.shouldHideBlogsNotSupportingDomains = config.shouldHideBlogsNotSupportingDomains
+        return dataSource
+    }
+
+    var allBlogs: [Blog] {
+        return dataSource.filteredBlogs
+    }
+
+    func updateSearchText(_ newText: String) {
+        if newText.isEmpty {
+            searchSites = allBlogs.compactMap(BlogListView.Site.init)
+        } else {
+            searchSites = allBlogs
+                .filter {
+                    $0.url?.lowercased().contains(newText.lowercased()) == true
+                    || $0.title?.lowercased().contains(newText.lowercased()) == true
+                }
+                .compactMap(BlogListView.Site.init)
+        }
+    }
+
+    func togglePinnedSite(siteID: NSNumber?) {
+        guard let blog = allBlogs.first(where: { $0.dotComID == siteID }) else {
+            return
+        }
+
+        blog.pinnedDate = blog.pinnedDate == nil ? Date() : nil
+        pinnedSites = Self.filteredPinnedSites(allBlogs: allBlogs)
+
+        let beforeRecentsCount = recentSites.count
+        recentSites = Self.filteredRecentSites(allBlogs: allBlogs)
+
+        if recentSites.count == beforeRecentsCount {
+            allRemainingSites = Self.filteredAllRemainingSites(allBlogs: allBlogs)
+        }
+        contextManager.saveContextAndWait(contextManager.mainContext)
+    }
+
+    func siteSelected(siteID: NSNumber?) {
+        guard let blog = allBlogs.first(where: { $0.dotComID == siteID }) else {
+            return
+        }
+
+        blog.lastUsed = Date()
+        recentSites = Self.filteredRecentSites(allBlogs: allBlogs)
+        contextManager.saveContextAndWait(contextManager.mainContext)
+    }
+
+    private static func filteredAllRemainingSites(allBlogs: [Blog]) -> [BlogListView.Site] {
+        allBlogs.filter({ $0.pinnedDate == nil && $0.lastUsed == nil }).compactMap(BlogListView.Site.init)
+    }
+
+    private static func filteredRecentSites(allBlogs: [Blog]) -> [BlogListView.Site] {
+        allBlogs
+            .filter({ $0.pinnedDate == nil && $0.lastUsed != nil })
+            .sorted(by: { $0.lastUsed! > $1.lastUsed! }) // Force-unwrapping due to the null check on line above
+            .prefix(8)
+            .compactMap(BlogListView.Site.init)
+    }
+
+    private static func filteredPinnedSites(allBlogs: [Blog]) -> [BlogListView.Site] {
+        allBlogs
+            .filter({ $0.pinnedDate != nil })
+            .sorted(by: { $0.pinnedDate! > $1.pinnedDate! }) // Force-unwrapping due to the null check on line above
+            .compactMap(BlogListView.Site.init)
+    }
+
+    private func selectedBlog() -> Blog? {
+        return RootViewCoordinator.sharedPresenter.currentOrLastBlog()
+    }
```

Comment:
```suggestion
```

Reply:
Resolved in [ad5f5c2](https://github.com/wordpress-mobile/WordPress-iOS/pull/22833/commits/ad5f5c26cb86249c900e9ccc9ef8e6f789f3655d)